LLama/LLamaExecutorBase.cs

```diff
@@ -315,6 +315,34 @@ public virtual async IAsyncEnumerable<string> InferAsync(string text, IInference
             }
         }
 
+        /// <summary>
+        /// Asynchronously runs a prompt through the model to compute KV cache without generating any new tokens.
+        /// </summary>
+        /// <param name="prompt">Prompt to process</param>
+        /// <param name="cancellationToken">A cancellation token</param>
+        /// <returns></returns>
+        public virtual async Task AddPromptAsync(string prompt, CancellationToken cancellationToken = default)
+        {
+            var inferenceParams = new InferenceParams
+            {
+                MaxTokens = 0
+            };
+            var args = new InferStateArgs
+            {
+                Antiprompts = new List<string>(),
+                RemainedTokens = 0,
+                ReturnValue = false,
+                WaitForInput = true,
+                NeedToSaveSession = false
+            };
+
+            await PreprocessInputs(prompt, args);
+            // First run adds the prompt to the _embeds
+            await InferInternal(inferenceParams, args);
+            // Second run puts it through decode
+            await InferInternal(inferenceParams, args);
```

Comment:
I'm not sure if it leads to infinite generation if no correct anti-prompt is set. Could you please confirm that?

Reply:
I set `WaitForInput = true` in inference args from the start which supposedly stops it from generating any tokens at all, from what I understand. (It's set to true when antiprompts are encountered in the generation implementation).